components/kyma-environment-broker/internal/process/provisioning/create_runtime.go

```diff
@@ -0,0 +1,111 @@
+package provisioning
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/provisioner"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-incubator/compass/components/provisioner/pkg/gqlschema"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+// Config holds all configurations connected with Provisioner API
+type Config struct {
+	URL             string
+	GCPSecretName   string
+	AzureSecretName string
+	AWSSecretName   string
+}
+
+type CreateRuntimeStep struct {
+	operationStorage  storage.Operations
+	instanceStorage   storage.Instances
+	builderFactory    InputBuilderForPlan
+	provisioningCfg   Config
+	provisionerClient provisioner.Client
+}
+
+func NewCreateRuntimeStep(os storage.Operations, is storage.Instances, builderFactory InputBuilderForPlan, cfg Config, cli provisioner.Client) *CreateRuntimeStep {
+	return &CreateRuntimeStep{
+		operationStorage:  os,
+		instanceStorage:   is,
+		builderFactory:    builderFactory,
+		provisioningCfg:   cfg,
+		provisionerClient: cli,
+	}
+}
+
+func (s *CreateRuntimeStep) Run(operation *internal.ProvisioningOperation) (error, time.Duration) {
+	// TODO make sure that the process has not already been activated
+	// TODO mark operation as failed if error is permanent
+
+	pp, err := operation.GetProvisioningParameters()
+	if err != nil {
+		return errors.Wrap(err, "while getting provisioning parameters from operation"), 0
+	}
+	rawParameters, err := json.Marshal(pp.Parameters)
+	if err != nil {
+		return errors.Wrap(err, "while marshaling instance parameters"), 0
+	}
+
+	input, err := s.createProvisionInput(pp)
```

Comment:
in the near future we will need to pass here also other fields from ProvisioningOperation instance